pkg/manager/manager.go

```diff
@@ -259,8 +259,20 @@ type Options struct {
 	// MetricsBindAddress is the TCP address that the controller should bind to
 	// for serving prometheus metrics.
 	// It can be set to "0" to disable the metrics serving.
+	//
+	// Per default metrics will be served via http.	If MetricsSecureServing is enabled
+	// metrics will be served via https.
 	MetricsBindAddress string
```

Comment:
Can we maybe move these three in a `MetricsOpts` sub struct or such? We have a set of global options like `Port` and `Host` and `TLSOpts` and it would likely be clearer separated if the metric-specific settings are in their own sub struct

Reply:
Yup sounds fine. Do you think we should already go one step further and to the same as with the webhook.Server?

Basically have a MetricsServer option + provide a func to create the "default" MetricsServer with options

Reply:
Also slightly wondering if we should take the chance and rename the server as  the introduction of `AddMetricsExtraHandler` was already the first step in allowing to use this server for more than just our metrics endpoint

(just not sure if there is a good framing / name for that)

P.S. One use case I have for metricsextrahandlers for this server on the CAPI side is to add an additional handler to allow setting the log level at runtime like in k/k: https://nakamasato.medium.com/how-to-change-log-level-of-kubernetes-components-3cd0e15b86ea